tpl/time/time.go

```diff
@@ -27,17 +27,19 @@ import (
 )
 
 // New returns a new instance of the time-namespaced template functions.
-func New(translator locales.Translator, location *time.Location) *Namespace {
+func New(translator locales.Translator, location *time.Location, buildTime string) *Namespace {
```

Comment:
Would it make more sense to pass a `time.Time` value here, rather than a string to be re-parsed, so that the build times are consistent across a single run?

As a potential bonus, by doing so the `location` parameter may become redundant.

Reply:
> Would it make more sense to pass a time.Time value here, rather than a string to be re-parsed, so that the build times are consistent across a single run?

Yes, you are right. Fixed.

> As a potential bonus, by doing so the location parameter may become redundant.

I do not think so.
This `location` is determined by site lang settings.
It is possible that you would like to use this flag with a different location from the lang settings'

Reply:
I have looked at this, and agree that we do this.

In short, we need something that allows us to say:

```
`expiryDate.Before(htime.Now())` 
```

Or something.

We should discuss this before doing this, but since I'm really tempting to take the simplest approach here, which is:

1. Add a `htime` package (if not already there)
2. In that package, add a public `var NowOverride time.Time´ and a `Now` func ala:

```go
func Now() time.Now {
   if !NowOverride.IsZero {
      return NowOverride
   }
   return time.Now()
}
```

And use that func where applicable. 

This means that tests using this cannot use `t.Parallel`, but it is extremely easy to reason about.

Reply:
> Add a htime package (if not already there)
> In that package, add a public var NowOverride time.Time and a Now func ala:
> This means that tests using this cannot use t.Parallel, but it is extremely easy to reason about.

Though I am concerning about test execution time and so on a little bit, probably it will be fine.

Reply:
> Though I am concerning about test execution time and so on a little bit, probably it will be fine.

I assumed there are some test cases to be fixed but I have found it is only `TestShouldBuild` in `page_test.go`.
pretty small.

Reply:
The test execution time should be negligible, and this i obviously the best choice from a pragmatic standpoint: Very simple to implement and reason about. Let's do it.

Reply:
Done!
Could you review this?

Reply:
Note that all the existing `t.Parallel` tests will still run in parallel, it's just tests that make changes to the global that needs to take extra care.

Reply:
You mean any other `t.Parallel` (which uses `NowOverride` internally) should be removed?

Reply:
No! That would kill the test build time. Tests that _changes the global time_ can not use `t.Parallel`, all existing tests are fine.

Reply:
ah...oops, I got what you meant finally.
Fixed.